pkg/steps/cluster_claim.go

```diff
@@ -180,7 +178,67 @@ func mutate(secret *corev1.Secret, name, namespace string) (*corev1.Secret, erro
 	}, nil
 }
 
-func ClusterClaimStep(as string, clusterClaim *api.ClusterClaim, hiveClient ctrlruntimeclient.Client, client loggingclient.LoggingClient, jobSpec *api.JobSpec) api.Step {
+var (
+	waitForClaim = func(client watchingclient.Client, ns, name string, timeout time.Duration) error {
+		logrus.WithFields(logrus.Fields{
+			"namespace": ns,
+			"name":      name,
+		}).Trace("Waiting for claim to be running.")
+
+		ctx := context.TODO()
+
+		lw := &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (object runtime.Object, e error) {
+				options.FieldSelector = fields.OneTermEqualSelector("metadata.name", name).String()
+				res := &hivev1.ClusterClaimList{}
+				return res, client.List(ctx, res, &ctrlruntimeclient.ListOptions{Namespace: ns, Raw: &options})
+			},
+			WatchFunc: func(options metav1.ListOptions) (i watch.Interface, e error) {
+				opts := &ctrlruntimeclient.ListOptions{
+					Namespace:     ns,
+					FieldSelector: fields.OneTermEqualSelector("metadata.name", name),
+					Raw:           &options,
+				}
+				return client.Watch(ctx, &hivev1.ClusterClaimList{}, opts)
+			},
+		}
+
+		// clusterClaims in this call here are always expected to exist in advance
+		var clusterClaimExistsPrecondition toolswatch.PreconditionFunc
+
+		waitForContainerStatus := func(event watch.Event) (bool, error) {
+			if event.Type == watch.Deleted {
+				return false, fmt.Errorf("clusterClaim/%v ns/%v was deleted", name, ns)
+			}
+			// the outer library will handle errors, we have no clusterClaim data to review in this case
+			if event.Type != watch.Added && event.Type != watch.Modified {
+				return false, nil
+			}
+			switch clusterClaim := event.Object.(type) {
+			case *hivev1.ClusterClaim:
+				// https://github.com/openshift/hive/blob/a535d29c4d2dc4f7f9bf3f30098c69d9334bee2e/apis/hive/v1/clusterclaim_types.go#L18-L22
+				for _, condition := range clusterClaim.Status.Conditions {
+					logrus.WithField("claim.Name", name).WithField("claim.Namespace", ns).
```

Comment:
This log line will just be spammy ... are you sure we need it? The logic for this watch processing is one `if` statement. It either works or it doesn't ;)

Reply:
Removed.